chrome/content/zotero/itemTree.jsx

```diff
@@ -978,6 +978,23 @@ var ItemTree = class ItemTree extends LibraryTree {
 			this.collapseAllRows();
 			return false;
 		}
+		// On arrowUp/down without modifiers in duplicates view, select the entire set
+		else if (this.collectionTreeRow.isDuplicates() && ["ArrowUp", "ArrowDown"].includes(event.key)
+				&& !event.shiftKey && !event.ctrlKey && !event.metaKey && !event.altKey) {
+			// Find the first row index before/after the current set
+			let selectedArray = Array.from(this.selection.selected).sort();
```

Comment:
Oh, good catch! I don't have that many duplicate items, so didn't run into the case when sorting wouldn't work right. 

Sorting probably wasn't the best approach to begin with... How about using `Math.min` and `Math.max`? That seems like the most straightforward and readable way to get max and min values in arrays.

Reply:
Duplicate sets don't have to be contiguous (it depends on the current sort column), so this logic isn't correct. It should just find the next row that's not part of the current set after the first row (for ArrowDown) or before the last row (for ArrowUp).

(Or I guess could use pivot row, if we actually set the pivot row to the row you click on, but that's probably more confusing, and after you use the arrow keys once the pivot row should be the first/last row anyway, so the effect would be the same after that.)

Reply:
Actually, maybe it does need to use a pivot row. Otherwise you can't go back and forth between two sets with up/down. For example:

![Screenshot 2024-11-06 at 8 10 14 AM](https://github.com/user-attachments/assets/defede98-72f1-45d5-a1a5-758e4d4112c7)

ArrowDown should select Table of Contents. But then if you press ArrowUp, if there's another Table of Contents above the top "Science ideated" row, it wouldn't necessarily select "Science ideated" again.

Reply:
And then I think the pivot row just needs to skip over any of the same set? So if the pivot row was the first row here (call it 1), ArrowDown would get you "Table of Contents" (2), "Zotero, OCR plugin…" (3), "Case–control matching…" (5, skipping 4), "Table of Contents" (6), "Science Ideated…" (7), and "Table of Contents" (9). ArrowUp would then go through those in reverse order.

Reply:
Oh, right. I didn't realize that one might have the table sorted by another column...

When you click on a row, or when you tab into the table, we do set `itemTree.selection.focused`, so that seems to be a nice candidate to use as the pivot. How does this behavior look? I added the red outline to which row is marked as `.selected`, just for the recording. 


https://github.com/user-attachments/assets/d51090c5-a79b-4dd2-92b5-f2be26e1df54